rails_programming/hotwire/turbo.md

```diff
@@ -0,0 +1,494 @@
+if you would like to learn more:
+
+1.  You can chain multiple `turbo_stream` actions in one `turbo_stream.erb` file, like if you wanted to append to one region and
+    update another region at the same time.
+
+2.  You don't *have* to create a separate layout file for your Stream formats. For instance, we could instead change the line in our controller to:
+
+    ~~~ruby
+    format.turbo_stream { render turbo_stream: turbo_stream.append('posts', @post) }
+    ~~~
+
+    However, [you shouldn't do this for anything complex or chained](https://github.com/hotwired/turbo-rails/issues/77#issuecomment-757349251).
+
+3.  You may have noticed that when you submit a new Post, the text box doesn't clear out! That's why Hotwire comes with Stimulus. In the next lesson, we will cover how to create a simple Stimulus controller to reset the new post form.
+
+### Turbo Native
+
+The final piece of Turbo is something that you don't need to know much about for this course, other than it exists. Turbo Native is useful for wrapping your application inside of a mobile-friendly shell. If you are interested in mobile development and how you can use Turbo with it, you may enjoy looking at the demos for [iOS](https://github.com/hotwired/turbo-ios/blob/main/Demo/README.md) and [Android](https://github.com/hotwired/turbo-android/blob/main/demo/README.md) phones.
+
+### Assignment
+
+<div class="lesson-content__panel" markdown="1">
+
+#### Single Page Applications
+
+1. Read sections 1, 2, and 3 of Bloomreach's [What Is A Single Page Application](https://www.bloomreach.com/en/blog/2018/07/what-is-a-single-page-application.html#whatssingle-page-application)
+   article
+
+#### Hotwire
+
+1. Watch the [Hotwire Demo Video](https://www.youtube.com/watch?v=eKY-QES1XQQ)
+    * We have only covered content up until the 5:40 mark, but you may continue watching past that point to become more familiar with
+      other aspects of Hotwire that we will be covering in upcoming lessons.
+    * The video is edited to be a very quick showcase. Don't worry about trying to pause and use this video as a tutorial. Just sit back
+      and use this demo to watch how Turbo Drive, Frames, & Streams come together visually.
+2. Skim through sections 1-4 of the [Turbo Handbook](https://turbo.hotwired.dev/handbook/introduction)
+    * This Handbook is written to be backend-agnostic, meaning that the code you will see is pure HTML and not Rails tags, but it still
+      is a useful resource for referencing how Turbo works!
+3. Take a quick glance at the Turbo-Rails gem [RubyDoc info page](https://www.rubydoc.info/gems/turbo-rails/1.0.0)
+    * This resource covers the Rails-specific syntaxes and tags you can use for Turbo. You don't need to read anything now, just know that
+      it exists so you can come back to it when you need to figure out how to use a specific piece of Turbo in your applications.
```

Comment:
Thanks for including that, but also sad, that this is the state of the Rails docs, there really should be at least one Rails guides on this.